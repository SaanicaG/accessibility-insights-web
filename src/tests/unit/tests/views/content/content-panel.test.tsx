// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { render } from '@testing-library/react';
import * as React from 'react';

import { ContentPage } from 'views/content/content-page';
import { ContentPanel } from 'views/content/content-panel';
import { ContentActionMessageCreator } from '../../../../../common/message-creators/content-action-message-creator';

describe('ContentPanel', () => {
    const content = {
        for: {
            testing: ContentPage.create(() => 'CONTENT FOR TESTING' as any),
        },
    };

    const applicationTitle = 'THE_APPLICATION_TITLE';
    const contentTitle = 'THE CONTENT TITLE';
    const deps = {
        textContent: {
            applicationTitle,
        },
        contentProvider: ContentPage.provider(content),
        contentActionMessageCreator: {} as any as ContentActionMessageCreator,
    };

    it('renders from content', () => {
        const renderResult = render(
            <ContentPanel
                deps={deps}
                content={content.for.testing}
                isOpen={true}
                contentTitle={contentTitle}
            />,
        );
        expect(renderResult.asFragment()).toMatchSnapshot();
    });

    it('renders from path', () => {
        const renderResult = render(
            <ContentPanel
                deps={deps}
                content={'for/testing'}
                isOpen={true}
                contentTitle={contentTitle}
            />,
        );
        expect(renderResult.asFragment()).toMatchSnapshot();
    });

    it('renders closed', () => {
        const renderResult = render(
            <ContentPanel
                deps={deps}
                content={'for/testing'}
                isOpen={false}
                contentTitle={contentTitle}
            />,
        );
        expect(renderResult.asFragment()).toMatchSnapshot();
    });
});
