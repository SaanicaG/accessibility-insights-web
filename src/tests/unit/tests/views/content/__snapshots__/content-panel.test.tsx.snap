// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentPanel renders closed 1`] = `<DocumentFragment />`;

exports[`ContentPanel renders from content 1`] = `
<DocumentFragment>
  <span
    class="ms-layer"
  />
</DocumentFragment>
`;

exports[`ContentPanel renders from path 1`] = `
<DocumentFragment>
  <span
    class="ms-layer"
  />
</DocumentFragment>
`;
